feat: learn the date object by building a date formatter (#50504)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/646e3d8a42296d01b0304b18.md

---
id: 646e3d8a42296d01b0304b18
title: Step 44
challengeType: 0
dashedName: step-44
---

# --description--

> You will add a `case` where the condition is `yyyy-mm-dd`. Inside this `case`, you will assign the constant `formattedDate` to the constant named `currentDateParagraph` using the `textContent` property.

Reviewer: ```suggestion
When the user selects the `Year, Month, Day` option, then the date format should reflect that. 

Add a `case` where the value is `yyyy-mm-dd`. Inside this `case`, assign the constant `formattedDate` to the constant named `currentDateParagraph` using the `textContent` property.
```

Author: added

---
id: 646e3d8a42296d01b0304b18
title: Step 17
challengeType: 0
dashedName: step-17
---

# --description--

When the user selects the `Year, Month, Day` option, then the date format should reflect that. 

> Add a `case` where the value is `yyyy-mm-dd`. Inside this `case`, assign the constant `formattedDate` to the constant named `currentDateParagraph` using the `textContent` property.

Reviewer: We will need to include a code example on how to add cases since this is the first time a switch statement is introduced. 

You will also need to mention that campers should add a break statement

Author: added

# --hints--

You need to use the `case` where the condition is `yyyy-mm-dd` to assign the constant `formattedDate` to the constant named `currentDateParagraph` using the `textContent` property.

```js

> assert(code.match(/case\s*['"]yyyy-mm-dd['"]\s*:\s*currentDateParagraph.textContent\s*=\s*formattedDate/g));

Reviewer 2: ```suggestion
assert(code.match(/case\s*(['"])yyyy-mm-dd\1\s*:\s*currentDateParagraph.textContent\s*=\s*formattedDate/g));
```
Same as before

      </div>
      <p id="current-date"></p>
    </main>
  </body>
  <script src="script.js"></script>
</html>
```

```css
*{
  margin: 0;
  padding: 0;
  box-sizing: border-box
}
:root {
  --dark-grey: #0a0a23;
  --white: #ffffff;
  --yellow: #f1be32;
}

body{
  background-color: var(--dark-grey);
  color: var(--white);
}

.title-container{
  margin: 10px 0;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
}

.title {
  font-size: 2.5rem;
}

.date-svg{
  width: 30px;
  height: 30px;
}

.divider {
  margin: auto;
  width: 150px;
  height: 10px;
  background-color: var(--yellow);
}

.dropdown-container {
  width: 100%;
  margin: 40px auto;
  position: relative;
}

select {
  display: block;
  margin: 20px auto;
  height: 50px;
  font-size: 100%;
  font-weight: bold;

  cursor: pointer;
  background-color: var(--white);
  border: none;
  color: var(--dark-grey);
  appearance: none;
  padding: 10px 38px 10px 10px;
}

.select-icon{
  position: absolute;
  top: 4px;
  right: 4px;
  pointer-events: none;
  width: 30px;
  height: 36px;
  padding-left: 4px;
}

#current-date{
  font-size: 2rem;
  text-align:center
}
```

```js

> const currentDateParagraph = document.getElementById("current-date")
> const dateOptionsSelectElement = document.getElementById("date-options")

Reviewer: ```suggestion
const currentDateParagraph = document.getElementById("current-date");
const dateOptionsSelectElement = document.getElementById("date-options");
```

const date = new Date();
const day = date.getDate();
const month = date.getMonth() + 1;
const year = date.getFullYear();
const hours = date.getHours();
const minutes = date.getMinutes();

> const formattedDate = `${day}-${month}-${year}`
> currentDateParagraph.textContent = formattedDate

Reviewer: ```suggestion
const formattedDate = `${day}-${month}-${year}`;
currentDateParagraph.textContent = formattedDate;
```

> dateOptionsSelectElement.addEventListener("change", ()=>{

Reviewer: ```suggestion
dateOptionsSelectElement.addEventListener("change", () => {
```

>   switch(dateOptionsSelectElement.value){

Reviewer: ```suggestion
  switch(dateOptionsSelectElement.value) {
```

--fcc-editable-region--

--fcc-editable-region--
    default: 
      currentDateParagraph.textContent = formattedDate

>     break

Reviewer: We should remove this break statement since it wasn't mentioned in the previous step and it is not needed since this is the default case 

```suggestion

```

  width: 100%;
  margin: 40px auto;
  position: relative;
}

select {
  display: block;
  margin: 20px auto;
  height: 50px;
  font-size: 100%;
  font-weight: bold;

  cursor: pointer;
  background-color: var(--white);
  border: none;
  color: var(--dark-grey);
  appearance: none;
  padding: 10px 38px 10px 10px;
}

.select-icon{
  position: absolute;
  top: 4px;
  right: 4px;
  pointer-events: none;
  width: 30px;
  height: 36px;
  padding-left: 4px;
}

#current-date{
  font-size: 2rem;
  text-align:center;
}
```

```js
const currentDateParagraph = document.getElementById("current-date");
const dateOptionsSelectElement = document.getElementById("date-options");

const date = new Date();
const day = date.getDate();
const month = date.getMonth() + 1;
const year = date.getFullYear();
const hours = date.getHours();
const minutes = date.getMinutes();

const formattedDate = `${day}-${month}-${year}`;
currentDateParagraph.textContent = formattedDate;

dateOptionsSelectElement.addEventListener("change", () => {
  switch(dateOptionsSelectElement.value) {
--fcc-editable-region--

--fcc-editable-region--
    default: 
      currentDateParagraph.textContent = formattedDate
  }

> })

Reviewer 3: ```suggestion
});
```
